packages/core/upload/server/services/upload.js

```diff
@@ -162,27 +162,30 @@ module.exports = ({ strapi }) => ({
       getDimensions,
       generateThumbnail,
       generateResponsiveFormats,
-      isSupportedImage,
+      isImage,
+      isOptimizableImage,
     } = getService('image-manipulation');
     await getService('provider').upload(fileData);
 
-    if (await isSupportedImage(fileData)) {
-      const thumbnailFile = await generateThumbnail(fileData);
-      if (thumbnailFile) {
-        await getService('provider').upload(thumbnailFile);
-        _.set(fileData, 'formats.thumbnail', thumbnailFile);
-      }
+    if (await isImage(fileData)) {
+      if (await isOptimizableImage(fileData)) {
```

Comment:
I think you could write it like:

```suggestion
    if (await isImage(fileData) && await isOptimizableImage(fileData)) {
```

if the first check will be falsy, the second won't be executed.